presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -151,16 +166,27 @@ private TTransport createTransport(HostAndPort address, Optional<String> delegat
         }
     }
 
-    private static KeyStore loadTrustStore(File trustStorePath)
+    private static KeyStore loadTrustStore(File trustStorePath, Optional<String> trustStorePassword)
             throws IOException, GeneralSecurityException
     {
-        List<X509Certificate> certificateChain = PemReader.readCertificateChain(trustStorePath);
-
         KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
-        trustStore.load(null, null);
-        for (X509Certificate certificate : certificateChain) {
-            X500Principal principal = certificate.getSubjectX500Principal();
-            trustStore.setCertificateEntry(principal.getName(), certificate);
+        try {
+            // attempt to read the trust store as a PEM file
+            List<X509Certificate> certificateChain = PemReader.readCertificateChain(trustStorePath);
+            if (!certificateChain.isEmpty()) {
+                trustStore.load(null, null);
+                for (X509Certificate certificate : certificateChain) {
+                    X500Principal principal = certificate.getSubjectX500Principal();
+                    trustStore.setCertificateEntry(principal.getName(), certificate);
+                }
+                return trustStore;
+            }
+        }
+        catch (IOException | GeneralSecurityException ignored) {
```

Comment:
As well as the key store load, we can show the warning or error message here to notify the detail of `GeneralSecurityException` instead of ignoring.

Reply:
Added warning log.